pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -96,18 +104,11 @@ func NewBuildGCSResource(r *PipelineResource) (*BuildGCSResource, error) {
 	}, nil
 }
 
-// GetName returns the name of the resource
-func (s BuildGCSResource) GetName() string {
-	return s.Name
-}
-
-// GetType returns the type of the resource, in this case "storage"
-func (s BuildGCSResource) GetType() PipelineResourceType {
-	return PipelineResourceTypeStorage
-}
-
-// GetSecretParams returns the resource secret params
-func (s *BuildGCSResource) GetSecretParams() []SecretParam { return nil }
+func (s BuildGCSResource) GetName() string                                         { return s.Name }
+func (s BuildGCSResource) GetType() PipelineResourceType                           { return PipelineResourceTypeStorage }
+func (s *BuildGCSResource) GetSecretParams() []SecretParam                         { return nil }
+func (s *BuildGCSResource) GetUploadSteps(string) ([]Step, error)                  { return nil, nil }
+func (s *BuildGCSResource) GetUploadVolumeSpec(*TaskSpec) ([]corev1.Volume, error) { return nil, nil }
```

Comment:
can we have docstrings for these? 🙏  (also note the changes in https://github.com/tektoncd/pipeline/pull/1188 that may or may not conflict with this)

Reply:
#1188 got closed without merging, it would have let me delete a lot of these. 😢 

Doc comments for most of them would be mostly meaningless, and only really useful to people perusing the godoc for some reason. Documenting in `docs/resources.md` seems much more useful and end-user-focused.

WDYT?

Reply:
> it would have let me delete a lot of these

😅 i think @dlorenc still has something planned!

> Doc comments for most of them would be mostly meaningless, and only really useful to people perusing the godoc for some reason. Documenting in docs/resources.md seems much more useful and end-user-focused.

I personally think the docs are worth it - also for reviewers that want to understand stuff like "why doesn't this return a volume spec?" and "why no secret params?"

one of the reasons it probably feels especially futile here is b/c there is something a bit off about this interface, which is why you're implementing functions that do nothing and why #1188 was trying to make it so you had to do less of that. I think we'll get to a point soon where authors of these resources will only be implementing functions that are actually useful for their resources and the docstrings will be useful.

i also opened https://github.com/tektoncd/plumbing/issues/64 to add linting for this... 😈

Reply:
`// GetUploadVolumeSpec returns nil.` <--This could also be an opportunity to say a bit more, even just like "GetUploadVolumeSpec doesn't return any volumes because no extra volumes are required for gcs resources" 😇 I agree that just repeating what the code says isn't super useful

Reply:
Done